lib/iris/_merge.py

```diff
@@ -1228,17 +1227,23 @@ def merge(self, unique=True):
                 data = self._skeletons[group[offset]].data
                 # Ensure the data is represented as a dask array and
                 # slot that array into the stack.
-                if is_lazy_data(data):
+                if iris.util.is_dask_array(data):
                     all_have_data = False
                 else:
-                    data = as_lazy_data(data)
+                    if isinstance(data, np.ma.MaskedArray):
+                        if np.ma.is_masked(data):
+                            if data.dtype.kind == 'i':
+                                data = data.astype('f8')
+                            data[data.mask] = np.nan
+                        data = data.data
+                    data = da.from_array(data, chunks=data.shape)
                 stack[nd_index] = data
 
             merged_data = _multidim_daskstack(stack)
             if all_have_data:
                 # All inputs were concrete, so turn the result back into a
                 # normal array.
-                merged_data = as_concrete_data(merged_data)
+                merged_data = merged_data.compute()
```

Comment:
@marqh `dask` is pretty greedy, and why not. By default `dask.array` uses the `dask.threaded.get` scheduler and will also default to using as many worker as there are cores.

I'd suggest that whenever we use `compute` we factor in a bit of grace and not handover **all** of the cores to `dask`. For me, a sensible *initial* default might be something like `multiprocessing.cpu_count() - 1`. 

We'll need to think (perhaps later once the dust settles) about how we pass options down into dask to control the scheduler of choice, see [here](http://dask.pydata.org/en/stable/scheduler-overview.html#configuring-the-schedulers), which may be relevant for those users targeting clusters and opting to use `distributed`.

For now, I'd suggest:
```python
import multiprocessing
...
    merged_data = merged_data.compute(num_workers=multiprocessing.cpu_count() - 1)
```